modules/tinymce/src/core/test/ts/webdriver/keyboard/IframeTabfocusTest.ts

```diff
@@ -1,37 +1,74 @@
-import { SugarDocument } from '@ephox/sugar';
-import { TinyHooks } from '@ephox/wrap-mcagar';
+import { FocusTools, RealKeys, RealMouse } from '@ephox/agar';
+import { after, afterEach, before, context, describe, it } from '@ephox/bedrock-client';
+import { Class, SugarDocument } from '@ephox/sugar';
+import { TinyDom, TinyHooks } from '@ephox/wrap-mcagar';
+import { assert } from 'chai';
 
 import Editor from 'tinymce/core/api/Editor';
 
 describe('webdriver.tinymce.core.keyboard.IframeTabfocusTest', () => {
-  const hook = TinyHooks.bddSetupLight<Editor>({
-    base_url: '/project/tinymce/js/tinymce',
-    iframe_attrs: { // using this instead of tabindex on the target to check custom attrs don't get clobbered
-      tabindex: '1'
-    }
-  }, [ ], true);
-
-  before(() => {
-    const editor = hook.editor();
-    const container = editor.getContainer();
-    const inputElem = document.createElement('input');
-    inputElem.id = 'inputElem';
-    inputElem.tabIndex = 2;
-    container.parentNode?.insertBefore(inputElem, container);
-  });
+  context('Focus shift on pressing tab', () => {
+    const hook = TinyHooks.bddSetupLight<Editor>({
+      base_url: '/project/tinymce/js/tinymce',
+      iframe_attrs: { // using this instead of tabindex on the target to check custom attrs don't get clobbered
+        tabindex: '1'
+      }
+    }, [ ], true);
+
+    before(() => {
+      const editor = hook.editor();
+      const container = editor.getContainer();
+      const inputElem = document.createElement('input');
+      inputElem.id = 'inputElem';
+      inputElem.tabIndex = 2;
+      container.parentNode?.insertBefore(inputElem, container);
+    });
 
-  after(() => {
-    const inputElem = document.getElementById('inputElem');
-    if (inputElem) {
-      inputElem.parentNode?.removeChild(inputElem);
-    }
+    after(() => {
+      const inputElem = document.getElementById('inputElem');
+      if (inputElem) {
+        inputElem.parentNode?.removeChild(inputElem);
+      }
+    });
+
+    it('TINY-8315: Add an input field outside the editor, focus on the editor, press the tab key and assert focus shifts to the input field', async () => {
+      FocusTools.isOnSelector('iframe is focused', SugarDocument.getDocument(), 'iframe');
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.text('\t') ]);
+      await FocusTools.pTryOnSelector('Wait for focus to be on input', SugarDocument.getDocument(), '#inputElem');
+    });
   });
 
-  it('TINY-8315: Add an input field outside the editor, focus on the editor, press the tab key and assert focus shifts to the input field', async () => {
-    FocusTools.isOnSelector('iframe is focused', SugarDocument.getDocument(), 'iframe');
-    await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.text('\t') ]);
-    await FocusTools.pTryOnSelector('Wait for focus to be on input', SugarDocument.getDocument(), '#inputElem');
+  context('Highlight editor content area on focus', () => {
```

Comment:
This could be tested without using webdriver since the focus border should be available on api calls like editor.focus() also it should also test the default that the does not have a focus border when the option is not specified.

I think we should only use webdriver tests when really necessary. So I would make a separate test for just the api focus tests. I guess we could still keep this one so make sure that keyboard focus does the right thing as well.

Reply:
I added API tests for the new option